pkg/3scale/amp/operator/upgrade.go

```diff
@@ -38,6 +41,14 @@ func (u *UpgradeApiManager) Upgrade() (reconcile.Result, error) {
 		return res, nil
 	}
 
+	res, err = u.upgradeBackendRouteEnv()
```

Comment:
I think we should upgrade the routes before changing the images.

Otherwise the new code that makes use of the environment variable will be effective before we change the route and currently we use an incorrect one.